runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContent.java

```diff
@@ -133,10 +137,22 @@ private void setNativeContent(long newNativeContent) {
         assert mNativeContent == 0 && mCleanupReference == null && mContentViewCore == null;
 
         // Initialize ContentViewRenderView
-        boolean animated = XWalkPreferencesInternal.getValue(
+        boolean animated;
+        if (attrs == null) {
+            animated = XWalkPreferencesInternal.getValue(
                 XWalkPreferencesInternal.ANIMATABLE_XWALK_VIEW);
+        } else {
+            // This means developer wants to assign the CompositingSurfaceType for a
+            // single XWalkView, "animatable" attribute of this namespace must be defined
+            // in activity.xml, while an attrs.xml that to define the type of "animatable" is
+            // also needed under res/values/
+            final String namespace = "http://schemas.android.com/apk/res-auto";
+            animated = attrs.getAttributeBooleanValue(namespace, "animatable", false);
```

Comment:
Nice, but per @axinging 's suggestion, it is to deprecate this method:
 <ol><li>[Deprecated]Set preference key
+ * {@link XWalkPreferencesInternal#ANIMATABLE_XWALK_VIEW} to True to use TextureView,